File: connected-devices-python workspace/iot-device/apps/labs/common/ActuatorData.py
'''
Created on Feb 1, 2019

@author: hkalyan
'''
import os
from datetime import datetime

COMMAND_OFF = 0
COMMAND_ON = 1
COMMAND_SET = 2
COMMAND_RESET = 3
STATUS_IDLE = 0
STATUS_ACTIVE = 1
ERROR_OK = 0
ERROR_COMMAND_FAILED = 1
ERROR_NON_RESPONSIBLE = -1

'''
This class is used for setting the various parameter of the Actuator
and return the compiled output data in a string format
'''
class ActuatorData():

    timeStamp = None
    name = 'Not set'
    hasError = False
    command = 0
    errCode = 0
    statusCode = 0
    stateData = None
    val = 0.0
    
    '''
    This is the constructor of the Actuator Data Class
    It has a function named UpdateTimeStamp()
    '''
    def __init__(self):
        self.updateTimeStamp()
        
    '''
    This will overwrite whatever value is currently stored in command.
    
    @return: Returns the command
    ''' 
    def getCommand(self):
            return self.command
    
    '''
    This will overwrite whatever value is currently stored in name.
    
    @return: Returns the name
    '''
    def getName(self):
        return self.name
    
    '''
    This will overwrite whatever value is currently stored in stateData.
    
    @return: Returns the stateData
    '''
    def getStateData(self):
        return self.stateData
    
    '''
    This will overwrite whatever value is currently stored in statusCode.
    
    @return: Returns the statusCode
    '''
    def getStatusCode(self):
        return self.statusCode
    
    '''
    This will overwrite whatever value is currently stored in ErrorCode.
    
    @return: Returns the ErrorCode
    '''
    def getErrorCode(self):
        return self.errCode
    
    '''
    This will overwrite whatever value is currently stored in value.
    
    @return: Returns the value
    '''
    def getValue(self):
        return self.val
    
    '''
    This will overwrite whatever value is currently stored in hasError.
    
    @return: Returns the hasError 
    '''
    def hasError(self):
        return self.hasError
    
    '''
    This will store the command.
    '''   
    def setCommand(self, command):
        self.command = command
    
    '''
    This will store the name.
    '''   
    def setName(self, name):
        self.name = name
    
    '''
    This will store the StateData.
    ''' 
    def setStateData(self, stateData):
        self.stateData = stateData
    
    '''
    This will store the statusCode.
    '''    
    def setStatusCode(self, statusCode):
        self.statusCode = statusCode
    
    '''
    This will store the ErrorCode.
    The hasError flag is set to true if errCode flag is true else hasErr is false
    '''    
    def setErrorCode(self, errCode):
        self.errCode = errCode
    
        if (self.errCode != 0):
            self.hasError = True
        else:
            self.hasError = False
    
    '''
    This will store the value.
    '''       
    def setValue(self, val):
        self.val = val
    
    '''
    This function updates the current data using the parameter passed
    
    @param data: The actuator data is passed and actuator variable are stored 
    ''' 
    def updateData(self, data):
        self.command = data.getCommand()
        self.statusCode = data.getStatusCode()
        self.errCode = data.getErrorCode()
        self.stateData = data.getStateData()
        self.val = data.getValue()
    
    '''
    This function stores the current date time to a container
    ''' 
    def updateTimeStamp(self):
        self.timeStamp = str(datetime.now())
    
    '''
    This funtion does object to string of this class
    Contains a simple container which has a string of all the data to be
    printed.
    
    @return: customStr: Returns the the string container
    '''
    def __str__(self):
        customStr = \
        str(self.name + ':' + \
        os.linesep + '\tTime: ' + self.timeStamp + \
        os.linesep + '\tCommand: ' + str(self.command) + \
        os.linesep + '\tStatus Code: ' + str(self.statusCode) + \
        os.linesep + '\tError Code: ' + str(self.errCode) + \
        os.linesep + '\tState Data: ' + str(self.stateData) + \
        os.linesep + '\tValue: ' + str(self.val))
        return customStr

File: connected-devices-java workspace/iot-gateway/src/main/java/schooldomain/studentname/connecteddevices/common/ActuatorData.java
/**
 *Created on Feb 16, 2019
 *
 *@author hkalyan
 */
package schooldomain.studentname.connecteddevices.common;

/**
*This class is used for setting the various parameter of the actuator
*and return the compiled output data in a string format
*/
public class ActuatorData {
	private String time;
	private String name;
	private boolean hasError = false;
	private int command = 0;
	private int errCode = 0;
	private int statusCode = 0;
	private  String stateData;
	private double val = 0.0;
	private int sampleCount = 0;
	
	/**
	 *Constructor for ActuatorData.
	 *
     *@param name The name of the data which is compiled.
     *@param time The time at which the data is generated
     *@param val The val that is passed to trigger an event
     *@param sampleCount The counter which keeps adding 
	 */
	public ActuatorData(String time, String name, double val, int sampleCount) {
		super();
		this.time = time;
		this.name = name;
		this.val = val;
		this.sampleCount = sampleCount;
	}

	/**
	 *Getters and setters of this class variables
	 */
	public String getTime() {
		return time;
	}


	public void setTime(String time) {
		this.time = time;
	}


	public String getName() {
		return name;
	}


	public void setName(String name) {
		this.name = name;
	}


	public boolean isHasError() {
		return hasError;
	}


	public void setHasError(boolean hasError) {
		this.hasError = hasError;
	}


	public int getCommand() {
		return command;
	}


	public void setCommand(int command) {
		this.command = command;
	}


	public int getErrCode() {
		return errCode;
	}


	public void setErrCode(int errCode) {
		this.errCode = errCode;
	}


	public int getStatusCode() {
		return statusCode;
	}


	public void setStatusCode(int statusCode) {
		this.statusCode = statusCode;
	}


	public String getStateData() {
		return stateData;
	}


	public void setStateData(String stateData) {
		this.stateData = stateData;
	}


	public double getVal() {
		return val;
	}


	public void setVal(double val) {
		this.val = val;
	}


	public int getSampleCount() {
		return sampleCount;
	}


	public void setSampleCount(int sampleCount) {
		this.sampleCount = sampleCount;
	}

	/**
	 *This method will return the string representation of the object
	 *
	 *@return str This is the string representation of the object of class ActuatorData
	 */
	@Override
	public String toString() {
		return "ActuatorData [time=" + this.getTime() + ", name=" + this.getName() + ", val=" + this.getVal() + ", sampleCount=" + this.getSampleCount() + "]";
	}
	
	/**
	 *This will increase the sampleCount value by 1
	 */
	public void addValue()
	{
		this.setSampleCount(this.getSampleCount()+1);
		
	}
}
